packages/node_modules/@webex/plugin-devicemanager/src/devicemanager.js

```diff
@@ -0,0 +1,503 @@
+        this.logger.info('DeviceManager#requestPin: space is not PIN challenge capable, probably already occupied, will still return device info');
+        return this.spark.internal.lyra.space.get(space)
+          .then(() => Promise.resolve(dev));
+      })
+      .catch((err) => {
+        this.logger.error('DeviceManager#requestPin: device failed PIN challenge request', err);
+        return Promise.reject(err);
+      });
+  },
+
+  /**
+  * pairs the device with the user (manual pairing), also adds it to
+  * user's recents list for subsequent fetches.
+  * similar to space.join()
+  * @param {object} options
+  * @param {number} options.pin
+  * @returns {Promise<deviceInfo>}
+  */
+  pair(options = {}) {
+    if (!options.pin) {
+      this.logger.error('DeviceManager#pair: options.pin is required');
+      return Promise.reject(new Error('DeviceManager#pair: options.pin is required'));
+    }
+    if (this._devicePendingPinChallenge) {
+      const space = {id: this._devicePendingPinChallenge.identity.id, url: `/spaces/${this._devicePendingPinChallenge.identity.id}`};
+
+      return this.spark.internal.lyra.space.join(space, {
+        passType: 'PIN_ANSWER',
+        data: options.pin
+      })
+        .then(() => this.upsert(this._devicePendingPinChallenge))
+        .catch((err) => {
+          this.logger.error('DeviceManager#pair: incorrect PIN, unable to pair ', err);
+          return Promise.reject(err);
+        });
+    }
+    this.logger.error('DeviceManager#pair: no device to pair');
+    return Promise.reject(new Error('DeviceManager#pair: no device to pair'));
+  },
+
+  /**
+  * unpairs the device with the user (manual/ultrasonic pairing), but still
+  * keeps in the recents list/does not remove from Redis
+  * options.removeAllDevices will remove all associated devices to user
+  * similar to space.leave()
+  * @param {object} options
+  * @param {boolean} options.removeAllDevices
+  * @returns {Promise<deviceInfo>}
+  */
+  unpair(options = {}) {
+    if (this._pairedDevice) {
+      const space = {url: `/spaces/${this._pairedDevice.identity.id}`};
+      return this.spark.internal.lyra.space.leave(space, options)
+        .catch((err) => {
+          this.logger.error('DeviceManager#unpair: failed to remove device from Lyra', err);
+          return Promise.reject(err);
+        });
+    }
+    return Promise.reject(new Error('DeviceManager#unpair: no device to unpair'));
+  },
+
+  /**
+  * binds the space to the paired device (if supported)
+  * similar to space.bindConversation()
+  * @param {string} convoId
+  * @returns {Promise<deviceInfo>}
+  */
+  bindSpace(convoId) {
+    if (this._pairedDevice) {
+      // fetch the conversation if just the convoId is provided
+      // TODO: no need to fetch the conversation if already provided
+      const url = `${this.spark.internal.device.services.conversationServiceUrl}/conversations/${convoId}`;
+      return this.spark.internal.conversation.get({url}, {
```

Comment:
Is convo always loaded? What if I load the device plugin outside of `ciscospark`?

Reply:
done